rust/operator-binary/src/restart_controller/pod.rs

```diff
@@ -192,6 +190,58 @@ async fn reconcile(pod: Arc<PartialObjectMeta<Pod>>, ctx: Arc<Ctx>) -> Result<Ac
     }
 }
 
+/// Reports the result of reconciliation.
+///
+/// The Pod restart controller has special handling, as it produced lot's of error messages below.
+/// They are expected, as we intentionally use the `Evict` API to restart Pods before e.g. the
+/// certificate expires. We roll out PDBs by default. If we try to restart multiple Pods that are
+/// part of a PDB, we get this errors.
+/// Because of this, we don't emit an error for this case, but only product a INFO trace.
+///
+/// `ERROR stackable_operator::logging::controller: Failed to reconcile object controller.name="pod.restarter.commons.stackable.tech" error=reconciler for object Pod.v1./trino-worker-default-0.default failed error.sources=[failed to evict Pod, ApiError: Cannot evict pod as it would violate the pod's disruption budget.: TooManyRequests (ErrorResponse { status: "Failure", message: "Cannot evict pod as it would violate the pod's disruption budget.", reason: "TooManyRequests", code: 429 }), Cannot evict pod as it would violate the pod's disruption budget.: TooManyRequests]`
+#[allow(clippy::type_complexity)] // The result type complexity comes from kube-rs and is what it is
+async fn report_result(
+    result: Result<
+        (ObjectRef<PartialObjectMeta<Pod>>, Action),
+        controller::Error<Error, watcher::Error>,
+    >,
+    event_recorder: Arc<Recorder>,
+) {
+    if let Err(controller::Error::ReconcilerFailed(
+        Error::EvictPod {
+            source: evict_pod_error,
+        },
+        pod_ref,
+    )) = &result
+    {
+        const TOO_MANY_REQUESTS_HTTP_CODE: u16 = StatusCode::TOO_MANY_REQUESTS.as_u16();
+        // We can not blanket silence all 429 responses, as it could be something else.
+        // E.g. I have seen "storage is re-initializing" in the past.
+        const EVICT_ERROR_MESSAGE: &str =
+            "Cannot evict pod as it would violate the pod's disruption budget.";
+
+        if let kube::Error::Api(ErrorResponse {
+            code: TOO_MANY_REQUESTS_HTTP_CODE,
+            message: error_message,
+            ..
+        }) = evict_pod_error
+        // TODO: We need Rust 1.88 and 2024 edition for if-let-chains
+        // && error_message == EVICT_ERROR_MESSAGE
+        {
+            if error_message == EVICT_ERROR_MESSAGE {
+                tracing::info!(
+                    k8s.object_ref = %pod_ref,
```

Comment:
Per https://github.com/stackabletech/commons-operator/pull/372#discussion_r2309890835...

```suggestion
                    k8s.object.ref = %pod_ref,
```